be/src/agent/task_worker_pool.cpp

```diff
@@ -707,6 +708,13 @@ void TaskWorkerPool::_publish_version_worker_thread_callback() {
                 std::this_thread::sleep_for(std::chrono::seconds(1));
             }
         }
+        if (res == OLAP_ERR_ROWSET_VERSION_NOT_CONTINUOUS) {
+            std::unique_lock<std::mutex> worker_thread_lock(_worker_thread_lock);
+            // push back to task queue if publish version not continuous
+            // make sure publish version task run in order
+            _tasks.push_back(agent_task_req);
```

Comment:
Maybe there will be a dead loop. If current tablet is a bad tablet(it really has a missed version), you will push the task to task queue and not publish it. Clone process will failed.